python-avd/pyavd/_eos_designs/structured_config/network_services/router_ospf.py

```diff
@@ -20,85 +23,90 @@ class RouterOspfMixin(Protocol):
-                    if l3_interface.ospf.enabled:
-                        for node_index, node in enumerate(l3_interface.nodes):
-                            if node != self.shared_utils.hostname:
-                                continue
-
-                            ospf_interfaces.append(l3_interface.interfaces[node_index])
-
-                for svi in vrf.svis:
-                    if svi.ospf.enabled:
-                        interface_name = f"Vlan{svi.id}"
-                        ospf_interfaces.append(interface_name)
+                process = EosCliConfigGen.RouterOspf.ProcessIdsItem()
+                self._update_ospf_interface(process, vrf)
 
                 process_id = default(vrf.ospf.process_id, vrf.vrf_id)
                 if not process_id:
                     msg = f"Missing or invalid 'ospf.process_id' or 'vrf_id' under vrf '{vrf.name}"
                     raise AristaAvdInvalidInputsError(msg)
-
-                process = {
-                    "id": process_id,
-                    "vrf": vrf.name if vrf.name != "default" else None,
-                    "passive_interface_default": True,
-                    "router_id": self.get_vrf_router_id(vrf, tenant, vrf.ospf.router_id),
-                    "no_passive_interfaces": ospf_interfaces,
-                    "bfd_enable": vrf.ospf.bfd or None,  # Historic behavior is to only output if True.
-                    "max_lsa": vrf.ospf.max_lsa,
-                }
-
-                process_redistribute = {}
-
-                if vrf.ospf.redistribute_bgp.enabled:
-                    process_redistribute["bgp"] = {"enabled": True}
-                    if route_map := vrf.ospf.redistribute_bgp.route_map:
-                        process_redistribute["bgp"]["route_map"] = route_map
-
-                if vrf.ospf.redistribute_connected.enabled:
-                    process_redistribute["connected"] = {"enabled": True}
-                    if route_map := vrf.ospf.redistribute_connected.route_map:
-                        process_redistribute["connected"]["route_map"] = route_map
-
-                process["redistribute"] = process_redistribute or None
-
-                # Strip None values from process before adding to list
-                process = {key: value for key, value in process.items() if value is not None}
-
-                append_if_not_duplicate(
-                    list_of_dicts=ospf_processes,
-                    primary_key="id",
-                    new_dict=process,
-                    context="OSPF Processes defined under network services",
-                    context_keys=["id"],
-                )
-
+                process._update(id=process_id, passive_interface_default=True)
+
+                if vrf.name != "default":
+                    process.vrf = vrf.name
+                if vrf_router_id := self.get_vrf_router_id(vrf, tenant, vrf.ospf.router_id):
+                    process.router_id = vrf_router_id
```

Comment:
The new classes can accept a None value, so you don't need all these `if`s

Reply:
Fixed!